storm-core/src/jvm/org/apache/storm/executor/Executor.java

```diff
@@ -204,8 +204,8 @@ public static Executor mkExecutor(Map workerData, List<Long> executorId, Map<Str
                 throw Utils.wrapInRuntime(ex);
             }
         }
-        executor.init(idToTask);
 
+        executor.idToTask = idToTask;
```

Comment:
@satishd 
Constructor of Task needs executor so Executor should be created prior to construct idToTask.